File: create_tables/strat_unit.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table strat_unit';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table strat_unit (
        STRAT_NAME_SET_ID VARCHAR(40) PRIMARY KEY NOT NULL,
        STRAT_UNIT_ID VARCHAR(40) NOT NULL,
        ABBREVIATION VARCHAR(12),
        ACTIVE_IND VARCHAR(1),
        AREA_ID VARCHAR(40),
        AREA_TYPE VARCHAR(40),
        BUSINESS_ASSOCIATE_ID VARCHAR(40),
        CONFIDENCE_ID VARCHAR(40),
        CURRENT_STATUS_DATE TIMESTAMP(0),
        DESCRIPTION VARCHAR(240),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        FAULT_TYPE VARCHAR(40),
        FORM_CODE VARCHAR(20),
        GROUP_CODE VARCHAR(20),
        LONG_NAME VARCHAR(255),
        ORDINAL_AGE_CODE DECIMAL(15,5),
        PPDM_GUID VARCHAR(38),
        PREFERRED_IND VARCHAR(1),
        REMARK VARCHAR(2000),
        SHORT_NAME VARCHAR(30),
        SOURCE VARCHAR(40),
        STRAT_INTERPRET_METHOD VARCHAR(40),
        STRAT_STATUS VARCHAR(40),
        STRAT_TYPE VARCHAR(40),
        STRAT_UNIT_TYPE VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")

File: postgres_39/all_struct_with_default/StratUnit.go
package dto

type Strat_unit struct {
	Strat_name_set_id      string   `json:"strat_name_set_id" default:"source"`
	Strat_unit_id          string   `json:"strat_unit_id" default:"source"`
	Abbreviation           *string  `json:"abbreviation" default:""`
	Active_ind             *string  `json:"active_ind" default:""`
	Area_id                *string  `json:"area_id" default:""`
	Area_type              *string  `json:"area_type" default:""`
	Business_associate_id  *string  `json:"business_associate_id" default:""`
	Confidence_id          *string  `json:"confidence_id" default:""`
	Current_status_date    *string  `json:"current_status_date" default:""`
	Description            *string  `json:"description" default:""`
	Effective_date         *string  `json:"effective_date" default:""`
	Expiry_date            *string  `json:"expiry_date" default:""`
	Fault_type             *string  `json:"fault_type" default:""`
	Form_code              *string  `json:"form_code" default:""`
	Group_code             *string  `json:"group_code" default:""`
	Long_name              *string  `json:"long_name" default:""`
	Ordinal_age_code       *float64 `json:"ordinal_age_code" default:""`
	Ppdm_guid              *string  `json:"ppdm_guid" default:""`
	Preferred_ind          *string  `json:"preferred_ind" default:""`
	Remark                 *string  `json:"remark" default:""`
	Short_name             *string  `json:"short_name" default:""`
	Source                 *string  `json:"source" default:""`
	Strat_interpret_method *string  `json:"strat_interpret_method" default:""`
	Strat_status           *string  `json:"strat_status" default:""`
	Strat_type             *string  `json:"strat_type" default:""`
	Strat_unit_type        *string  `json:"strat_unit_type" default:""`
	Row_changed_by         *string  `json:"row_changed_by" default:""`
	Row_changed_date       *string  `json:"row_changed_date" default:""`
	Row_created_by         *string  `json:"row_created_by" default:""`
	Row_created_date       *string  `json:"row_created_date" default:""`
	Row_effective_date     *string  `json:"row_effective_date" default:""`
	Row_expiry_date        *string  `json:"row_expiry_date" default:""`
	Row_quality            *string  `json:"row_quality" default:""`
}
